### PR TITLE
CLI: Parse negative Isla term literals

### PR DESCRIPTION
**Pull request by febyeji**
- Add unary minus support to Isla string terms.
- Convert lexer failures into TOML errors.
- Cover negative string init and unsupported sign extension through CLI tests.

### CODE REVIEW

File: cli/lib/isla/bv_fns.ml [hunk 1]
```diff
@@ -119,9 +125,14 @@ let functions : (string * Fn_registry.fn_spec) list =
...
-            Litmus.Error.failwith
-              "exts is not support because of integer semantics"
+          (fun args ->
+            match args with
+            | [bits; len] ->
+                check_non_negative_arg "exts" "bits" bits;
+                let len = int_of_bit_arg "exts" "len" len in
+                if len = 0 || not (Z.testbit bits (len - 1)) then bits
+                else Z.sub bits (Z.shift_left Z.one len)
+            | _ -> Fn_registry.arity_error "exts" 2 (List.length args)
```
Comment (tperami) [inline cli/lib/isla/bv_fns.ml]: I'm not convinced this code is correct, and we found only one test using this function, so I would leave it disabled

File: cli/lib/isla/converter.ml [hunk 2]
```diff
@@ -93,15 +93,34 @@ let term_eval_context_path = function
...
+  let value = eval_term ~context ~resolve_sym value in
+  (* Memory values are converted with Z.to_bits. *)
+  if Z.sign value < 0 then
+    raise
+      (Term_eval_error
+         ( term_eval_context_path context,
+           Printf.sprintf "negative memory data is not allowed: %s"
+             (Z.format "%#x" value)
+         )
+      );
```
Comment (tperami) [inline cli/lib/isla/converter.ml]: It's very weird to have this check here but the one about final condition directly in `eval_term`. Probably both should be in `eval_term`

File: cli/tests/errors/positive-exts.litmus.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+arch = "AArch64"
+name = "positive-exts"
+
+[thread.0]
+init = { X0 = "exts(0xff, 8)" }
+code = "NOP"
+
+[final]
+assertion = "0:X0 = 0xffffffffffffffff"
```
Comment (tperami) [inline cli/tests/errors/positive-exts.litmus.toml]: This is not the correct semantic for `exts`, the integer argument is the length of the resulting bitvector in bitvector semantics, which can't emulate with integer semantics

**Review by tperami — CHANGES_REQUESTED** [review]
The semantics of `exts` is wrong and need to be purged. It cannot be implemented without bitvector semantics

Also, a code organization issue: mix-and-matching error checking at different level is confusing

File: cli/tests/errors/errors.t [hunk 4]
```diff
@@ -149,3 +149,74 @@ Isla function error in final assertion
...
+  $ archsem seq negative-init.litmus.toml
+  Test negative-init Allowed
+  States 1
+  0:X0=0xffffffffffffffff; 
+  Ok
+  Witnesses
+  Positive: 1 Negative: 0
+  Condition negative-init exists 0:X0=0xffffffffffffffff
+  Observation negative-init Always 1 0
+  Time negative-init 0.000
```
Comment (tperami) [inline cli/tests/errors/errors.t]: Also, ideally we shouldn't add success tests here. We can just add a test with negative inputs and functions on negative bitwise value (and for example using an `AND` instruction) to `cli/tests/arm/seq`

File: cli/tests/errors/errors.t [hunk 5]
```diff
@@ -38,20 +38,6 @@ Just architecture and nothing else
...
-  0:R0=0x110; 0:_PC=0x504; 
-  Ok
-  Witnesses
-  Positive: 1 Negative: 0
-  Condition EOR forall (0:_PC=0x504 /\ 0:R0=0x110)
-  Observation EOR Always 1 0
-  Time EOR 0.000
-  archsem: name warning:
-  File "name-mismatch.archsem.toml" has a different name that the internal name "EOR"
-
```
Comment (tperami) [inline cli/tests/errors/errors.t:54]: This one was to check we get the name mismatch warning, why did you remove it. It has nothing to do with this PR

File: cli/lib/isla/bv_fns.ml [hunk 6]
```diff
@@ -110,19 +115,11 @@ let functions : (string * Fn_registry.fn_spec) list =
...
       }
-    );
-    ( "exts",
-      { params = ["bits"; "len"];
-        eval =
-          (fun _ ->
-            Litmus.Error.failwith
-              "exts is not support because of integer semantics"
-          )
-      }
```
Comment (tperami) [inline cli/lib/isla/bv_fns.ml:126]: Why are you removing the error message?

File: cli/lib/isla/converter.ml [hunk 7]
```diff
@@ -93,15 +93,44 @@ let term_eval_context_path = function
...
+    match context with
+    (* Final constants stay raw Z.t; registers read via bv_unsigned. *)
+    | Final_assertion when Z.sign value < 0 ->
+        raise
+          (Term_eval_error
+             ( term_eval_context_path context,
+               Printf.sprintf "final assertion values must be non-negative: %s"
+                 (Z.format "%#x" value)
+             )
+          )
```
Comment (tperami) [inline cli/lib/isla/converter.ml]: Making a function to raise the error directly would avoid the boilerplate. Something like
```ocaml
eval_error context "final assertion values must be non-negative: %s" (Z.format "%#x" value)
```
